demonstrations/tutorial_classical_shadows.py

```diff
@@ -0,0 +1,708 @@
+#
+# To obtain a classical shadow using PennyLane, we add the ``calculate_classical_shadow``
+# function below.
+# This function obtains a classical shadow for the state prepared by the
+# ``circuit_template``.
+
+
+def calculate_classical_shadow(circuit_template, params, shadow_size, num_qubits):
+    """
+    Given a circuit, creates a collection of snapshots consisting of a bit string
+     and the index of a unitary operation.
+
+    Args:
+        circuit_template (function): A Pennylane QNode.
+        params (array): Circuit parameters.
+        shadow_size (int): The number of snapshots in the shadow.
+        num_qubits (int): The number of qubits in the circuit.
+
+    Returns:
+        Tuple of two numpy arrays. The first array contains measurement outcomes (-1, 1)
+        while the second array contains the index for the sampled Pauli's (0,1,2=x,y,z).
+        Each row of the arrays corresponds to a distinct snapshot or sample while each
+        column corresponds to a different qubit.
+    """
+    unitary_ensemble = [qml.PauliX, qml.PauliY, qml.PauliZ]
+
+    # sample random Pauli unitaries uniformly, where 0,1,2 = X,Y,Z
+    unitary_ids = np.random.randint(0, 3, size=(shadow_size, num_qubits))
+    outcomes = np.zeros((shadow_size, num_qubits))
+
+    for ns in range(shadow_size):
+        # for each snapshot, add a random Pauli observable at each location
+        obs = [unitary_ensemble[int(unitary_ids[ns, i])](i) for i in range(num_qubits)]
+        outcomes[ns, :] = circuit_template(params, observable=obs)
+
+    # combine the computational basis outcomes and the sampled unitaries
+    return (outcomes, unitary_ids)
+
+
+##############################################################################
+# As an example, we demonstrate how to use ``calculate_classical_shadow`` and
+# check its performance as the number of snapshots increases.
+# First, we will create a two-qubit device and a circuit that applies an
+# ``RY`` rotation to each qubit.
+
+num_qubits = 2
+
+# setting up a two-qubit device. Setting shots = 1 ensures that we only get a single measurement
+dev = qml.device("default.qubit", wires=num_qubits, shots=1)
+
+# simple circuit to prepare rho
+@qml.qnode(device=dev)
+def local_qubit_rotation_circuit(params, **kwargs):
+    observables = kwargs.pop("observable")
+    for w in dev.wires:
+        qml.RY(params[w], wires=w)
+
+    return [qml.expval(o) for o in observables]
+
+
+# arrays in which to collect data
+elapsed_times = []
+shadows = []
+
+# collecting shadows and elapsed times
+params = np.random.randn(2)
+for num_snapshots in [10, 100, 1000, 10000]:
+    start = time.time()
+    shadow = calculate_classical_shadow(
+        local_qubit_rotation_circuit, params, num_snapshots, num_qubits
+    )
+    elapsed_times.append(time.time() - start)
+    shadows.append(shadow)
+
+# printing out the smallest shadow as an example
+shadows[0][0]
+shadows[0][1]
```

Comment:
```suggestion
print(shadows[0][0])
print(shadows[0][1])
```
Otherwise only the second one actually shows up in the output.